components/InOutputFields.js

```diff
@@ -1,12 +1,71 @@
 import styled from "styled-components";
+import React, {useState} from "react";
+import {RiAddCircleFill} from "react-icons/ri";
+import {AiFillMinusCircle} from "react-icons/ai";
 
-export default function InOutputFields({change, setChange}) {
+export default function InOutputFields({
+  change,
+  setChange,
+  savedChanges,
+  setSavedChanges,
```

Comment:
I only learned recently that it's best practice not to pass setter functions as props. what you would do is write a function e.g. `function handleChange(param){setChange(param)}` and pass this one as `onChange={handleChange}`